### PR TITLE
Extracting message from Ticket

### PR DESCRIPTION
**Pull request by sntiwari1**
1. Trim HTML elements
2. Trim Signatures, Salutation from Tickets

### CODE REVIEW

File: frappedesk/rocketchat/trim_email.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        # removing html tags
+        recent_message = remove_html_tags(email_content)
+        # removing reply text
+        recent_message = remove_original_message(recent_message)
+    ## Using talon to remove signature from email
+    text, signature = extract_signature(recent_message)
+    ## Using custom function to remove original message from email
+    if text:
+        # removing signature and salutation
+        recent_message = get_body(text)
```
Comment (ashishnpbs) [inline frappedesk/rocketchat/trim_email.py:160]: if we do not get any text after our processing - return original email_content

File: frappedesk/rocketchat/rocketchat.py [hunk 2]
```diff
@@ -28,7 +29,7 @@
 }
 
 def sendMessages(msg, postData=data, headers=headers):
-	msg = re.sub('<[^<]+?>', '', msg)
+	msg = extract_original_message(msg)
```
Comment (ashishnpbs) [inline frappedesk/rocketchat/rocketchat.py:32]: only call if the message is bigger than our limits of 256 chars

File: frappedesk/rocketchat/trim_email.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
+from talon.signature.bruteforce import extract_signature
+import re
+
+def remove_original_message(email_content):
+    #regex = r"(?s)^\s*\bOn\b.*\nwrote:.*$"
```
Comment (ashishnpbs) [inline frappedesk/rocketchat/trim_email.py:5]: does this assume that the original message will always be in bottom -- what if the original message is at top?

File: frappedesk/rocketchat/trim_email.py (hunk 1 above)
Comment (ashishnpbs) [reply frappedesk/rocketchat/trim_email.py:160]: actually this should be the final return value of get_body -- if this is null -- return original content

we can also say that in any step - if the input is less than 256 chars - do not do anything

File: frappedesk/rocketchat/trim_email.py (hunk 3 above)
Comment (ashishnpbs) [reply frappedesk/rocketchat/trim_email.py:5]: As discussed - why to have two separate functions -- why not merge remove_original_message and get_reply_text?

File: frappedesk/rocketchat/trim_email.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    if not groups is None:
+        if 'reply_text' in groups.groupdict():
+            reply_text = groups.groupdict()["reply_text"]
+    return reply_text
+
+
+def get_signature(email_text):
+
+    #try not to have the signature be the very start of the message if we can avoid it
+    salutation = get_salutation(email_text)
```
Comment (ashishnpbs) [inline frappedesk/rocketchat/trim_email.py:47]: we already removed salutation

File: frappedesk/rocketchat/trim_email.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+                              "yours truly",
+                              "thanking You",
+                              "sent from my iphone"]
+    pattern = "(?P<signature>(" + "|".join(sig_opening_statements) + ")(.)*)"
+    groups = re.search(pattern, email_text, re.IGNORECASE + re.DOTALL)
+    signature = None
+    if groups:
+        if "signature" in groups.groupdict():
+            signature = groups.groupdict()["signature"]
+            reply_text = get_reply_text(email_text[email_text.find(signature):])
```
Comment (ashishnpbs) [inline frappedesk/rocketchat/trim_email.py:74]: we are calling get_reply text -- why again here
also should it be reverse ? email_text[:--]

File: frappedesk/rocketchat/trim_email.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+            tmp_sig = signature
+            for s in sig_opening_statements:
+                if tmp_sig.lower().find(s) == 0:
+                    tmp_sig = tmp_sig[len(s):]
+            groups = re.search(pattern, tmp_sig, re.IGNORECASE + re.DOTALL)
+            if groups: signature = groups.groupdict()["signature"]
+
+    #if no standard formatting has been provided (e.g. Regards, <name>),
+    #try a probabilistic approach by looking for phone numbers, names etc. to derive sig
+    if not signature:
```
Comment (ashishnpbs) [inline frappedesk/rocketchat/trim_email.py:87]: why is this code here?

File: frappedesk/rocketchat/trim_email.py [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+        #body_without_sig = get_body(email_text, check_signature=False)
+        pass
+
+    #check to see if the entire body of the message has been 'stolen' by the signature. If so, return no sig so body can have it.
+    body_without_sig = get_body(email_text, check_signature=False)
```
Comment (ashishnpbs) [inline frappedesk/rocketchat/trim_email.py:92]: why it is calling get_body?

**Review by ashishnpbs — COMMENTED** [review]
Not sure why we are calling the same function multiple times -- like get_reply_text and get_body -- please restructure

File: frappedesk/rocketchat/trim_email.py [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+    salutation_opening_statements = [
+                                     "hi",
+                                     "dear",
+                                     "to",
+                                     "hey",
+                                     "hello",
+                                     "thanks",
+                                     "good morning",
+                                     "good afternoon",
+                                     "good evening"]
```
Comment (jahnaviraj) [inline frappedesk/rocketchat/trim_email.py:81]: can we also add "respected" here?

**Pull request closed without merging**